filter_gen.py

```
@@ -0,0 +1,436 @@
+# Extension libraries.
+import numpy as np
+import scipy.signal
+
+type_print_form = {'lowpass': 'Low-Pass', 'highpass': 'High-Pass', 'bandpass': 'Band-Pass', 'bandstop': 'Band-Stop'}
+
+################################################################
+# Optionally generate plots of the filter properties.
+def make_plots(filename, sos, fs, order, freqs, name):
+    try:
+        import matplotlib.pyplot as plt
+    except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)